examples/Huggingface_Largemodels/Readme.md

```diff
@@ -0,0 +1,59 @@
+# Loading large Huggingface models with contrained resources using accelerate
+
+This document briefs on serving large HG models with limited resource using accelerate. This option can be activated with `low_cpu_mem_usage=True`. The model is first created on the Meta device (with empty weights) and the state dict is then loaded inside it (shard by shard in the case of a sharded checkpoint).
+
+### Step 1: Download model
+
+Login into huggingface hub with token by running the below command
+
+```bash
+huggingface-cli login
+```
+paste the token generated from huggingface hub.
+
+```bash
+python Download_model.py --model_name bigscience/bloom-7b1
+```
+The script prints the path where the model is downloaded as below.
+
+`model/models--bigscience-bloom-7b1/snapshots/5546055f03398095e385d7dc625e636cc8910bf2/`
+
+The downloaded model is around 14GB.
+
+### Step 2: Compress downloaded model
+
+**_NOTE:_** Install Zip cli tool
+
+Navigate to the path got from the above script. Here it is
+
+```bash
+cd model/models--bigscience-bloom-7b1/snapshots/5546055f03398095e385d7dc625e636cc8910bf2/
+zip -r /serve/examples/deepspeed_mii/model.zip *
+```
+
+### Step 3: Generate MAR file
+
+Navigate up to `Huggingface_Largemodels` directory.
+
+```bash
+torch-model-archiver --model-name bloom --version 1.0 --handler custom_handler.py --extra-files model.zip,setup_config.json -r requirements.txt
+```
+
+**__Note__**: Modifying setup_config.json
+- Enable `low_cpu_mem_usage` to use accelerate
+- Recommended `max_memory` in `setup_config.json` is the max size of shard.
```

Comment:
can you please clarify a bit why we set it to 10 GB, and if shards based on the size? whats is a good rule of thumb here?